src/compile/axis/encode.ts

```diff
@@ -67,13 +70,16 @@ export function labelAngle(axis: Axis, channel: Channel, fieldDef: FieldDef<stri
   }
   return undefined;
 }
-export function labelAlign(angle: number, orient: AxisOrient) {
-  if (angle && angle > 0) {
-    if (angle > 180) {
+
+export function labelAlign(angle: number, orient: AxisOrient): HorizontalAlign {
+  if (angle > 0) {
+    if (angle % 360 > 180) {
```

Comment:
Why not normalize the angle once?

Reply:
I think it's not relevant to this PR and it's not easy to see this large diff here so I'll just merge.